src/components/tooltip/styles.ts

```diff
@@ -1,85 +1,27 @@
+import { Strategy } from "@floating-ui/react";
 import styled, { css } from "styled-components";
 import theme from "../../theme";
 import { setColor } from "../../utils/setcolor";
 
-function setArrowPosition(position: string) {
-  switch (position) {
-    case "left":
-      return `
-        left: 0.5rem;
-        right: auto;
-        margin-left: 0;
-        `;
-    case "right":
-      return `
-        left: auto;
-        right: 0.5rem;
-        margin-left: 0;
-        `;
-    default:
-      return `
-        left: 50%;
-        margin-left: -0.25rem;
-        right:auto;
-      `;
-  }
-}
-
-export const TooltipWrapper = styled.div`
-  display: inline-block;
-`;
+export const DEFAULT_TOOLTIP_PADDING = 5;
 
 type StyledTooltipProps = {
-  position?: "left" | "right";
-  bottom?: boolean;
+  strategy: Strategy;
   status?: "danger" | "notification" | "success" | "warning";
-  minWidth?: string;
 };
 
+// Don't forward style prop, as we need to add positioning data and passing `style` prop would overwrite it
 export const StyledTooltip = styled.div<StyledTooltipProps>`
-  position: absolute;
-  min-width: ${({ minWidth }) => minWidth};
-  max-width: 16rem;
+  width: max-content;
+  /* We want to ensure that padding from 2 sides is accounted for, hence multiplication */
+  max-width: min(calc(100vw - ${DEFAULT_TOOLTIP_PADDING * 2}px), 16rem);
```

Comment:
It doesn't matter since nobody's gonna get maximal width but `vw` includes sidebar width, which we probably don't want to account here.

Reply:
I've added additional CSS variable to ensure it works every time now